context/app/api/vitessce.py

```diff
@@ -170,57 +136,98 @@
+            "type": "RASTER",
+        },
+        {
+            "rel_path": f"{CODDEX_SPRM_PATH}/{TILE_REGEX}.cells.json",
+            "fileType": "cells.json",
+            "type": "CELLS",
+        },
+        {
+            "rel_path": f"{CODDEX_SPRM_PATH}/{TILE_REGEX}.cell-sets.json",
+            "fileType": "cell-sets.json",
+            "type": "CELL-SETS",
+        },
+        {
+            "rel_path": f"{CODDEX_SPRM_PATH}/{TILE_REGEX}.clusters.json",
+            "fileType": "clusters.json",
+            "type": "EXPRESSION-MATRIX",
+        },
+    ],
 }
 
-IMAGE_ASSAYS = [CODEX_CYTOKIT, IMAGE_PYRAMID]
-TILED_ASSAYS = [CODEX_CYTOKIT]
+IMAGE_PYRAMID_PATH = "ometiff-pyramids"
+IMAGE_PYRAMID_CONFIG = {
+    "base_conf": IMAGING_ONLY,
+    # TODO: We can actually fetch height/width using a COG tiff library,
+    # but for now this will do.
+    "view": {"zoom": -6.5, "target": [15000, 15000, 0]},
+    "files_conf": [
+        {
+            "rel_path": re.escape(IMAGE_PYRAMID_PATH) + r"/.*\.ome\.tiff?$",
+            "fileType": "raster.json",
+            "type": "RASTER",
+        },
+    ],
+}
+
+SEQFISH_HYB_CYCLE_REGEX = r"(HybCycle_\d+|final_mRNA_background)"
+SEQFISH_NAME_REGEX = r"MMStack_Pos\d+\.ome\.tiff?"
+SEQFISH_REGEX = f"{IMAGE_PYRAMID_PATH}/{SEQFISH_HYB_CYCLE_REGEX}/{SEQFISH_NAME_REGEX}"
+
+OFFSETS_PATH = "output_offsets"
+
+
+class Const(object, metaclass=SingletonMetaClass):
+    """
+    This class builds some constant data structures based on calls to the type client.  It is a
+    singleton, which means only a single instance is ever created, so the tables are built only
+    once and reused as needed.
+    """
+    def __init__(self):
+        """
+        Initialize the type client and build the required tables, which will be accessed as
+        attributes of the singleton instance.
+        """
+        tc = TypeClient(current_app.config['TYPE_SERVICE_ENDPOINT'])
+        conf_lut = {}
+        path_lut = {}
+        sc_l = []
+        for assay in tc.iterAssays():
+            name = assay.name
+            hints = assay.vitessce_hints
+
+            if 'codex' in hints:
+                path_lut[name] = {"offsets": OFFSETS_PATH, "image": CODEX_TILE_PATH}
+            elif 'pyramid' in hints:
+                path_lut[name] = {"offsets": OFFSETS_PATH, "image": IMAGE_PYRAMID_PATH}
+
+            if 'is_sc' in hints:
+                sc_l.append(name)
+                if 'atac' in hints:
+                    conf_lut[name] = SCATAC_SEQ_CONFIG
+                elif 'rna' in hints:
+                    conf_lut[name] = SCRNA_SEQ_CONFIG
+            elif 'is_image' in hints:
+                if 'pyramid' in hints:
+                    conf_lut[name] = IMAGE_PYRAMID_CONFIG
+                elif 'codex' in hints:
+                    conf_lut[name] = CODEX_CYTOKIT_CONFIG
+
+        self.ASSAY_CONF_LOOKUP = conf_lut
+        self.IMAGING_PATHS = path_lut
+        self.SC_DATA_TYPES = sc_l
```

Comment:
I'm going to argue in favor of brevity here.  It's not at all clear to me that a direct assignment is clearer, and it's computationally less efficient because it requires an additional dereference.  An increase in clarity would outweigh that, but as I say, I'm not convinced.